Don't allow guideline selection, if they are hidden.

ollimeier opened the pull request:
Fixes #1760 

Hi @justvanrossum , this is a possible solution, but I am not sure if this is the right path.
My thinking here is, that when the visualizationLayersSettings for guidelines is set to be 'hidden', the guidelines should not be part of the general selection. But to check the status of the guidelines visualizationLayersSettings we need `editorController`, but this hasn't been part of `SceneModel`, yet. And I think it may has a reason??? 
Therefore this is just a proposal to get in a conversation with you to get some advice.

justvanrossum commented:
Giving SceneController a reference to EditorController is definitely not the right solution, as it breaks the abstraction. Will look at it later.

ollimeier commented:
> > Giving SceneController a reference to EditorController is definitely not the right solution, as it breaks the abstraction. Will look at it later.
> 
> Since you only need `editorController.visualizationLayersSettings`, please pass _that_ instead of the entire `editorController` object. This leaves the abstraction graph intact.
> 
> (Before you say, "we could do one step further, and pass `editorController.visualizationLayersSettings.model`": yes, but not doing that will make it trivial for the `SceneModel` to subscribe to changes in the visualization layer settings, which can be useful in the future. So no.)

Thank you very much for your feedback. I changed it as requested. I also updated the `doSelectAllNone()`, so that you cannot select guidelines via shortcuts, when they are hidden. The PR is now 'Ready for review'.
